src/main/java/com/ericsson/ei/handlers/RmqHandler.java

```diff
@@ -171,16 +198,59 @@ Binding binding() {
     }
 
     @Bean
-    public List<Binding> bindings() {
+    public List<Binding> bindings(){
         final String[] bingingKeysArray = splitBindingKeys(bindingKeys);
         final List<Binding> bindingList = new ArrayList<>();
         for (final String bindingKey : bingingKeysArray) {
             bindingList.add(BindingBuilder.bind(externalQueue()).to(exchange()).with(bindingKey));
         }
+        deleteBindings(bingingKeysArray,bindingList);
         return bindingList;
     }
 
-    @Bean
+    private void deleteBindings(String[] bingingKeysArray, List<Binding> bindingList){
+    // Creating BindingKeys Collection in mongoDB
+        String bindingKeyCollectionName = "binding_keys";
+        ArrayList<String> allDocuments = mongoDBHandler.getAllDocuments(bindingKeysDataBaseName, bindingKeyCollectionName);
+        ArrayList<String> existingBindingsData = new ArrayList<String>();
+        if(!allDocuments.isEmpty()){
+            for(String bindings : allDocuments) {
+                JSONObject bindingObj = new JSONObject(bindings);
+                final String mongoDbBindings = bindingObj.getString("bindingKeys");
+                String condition = "{\"bindingKeys\": /.*" + mongoDbBindings + "/}";
+                    if(!Arrays.asList(bingingKeysArray).contains(mongoDbBindings)){
+                        String destinationDB = bindingObj.getString("destination");
+                        String exchangeDB = bindingObj.getString("exchange");
+                        // Binding the old binding key and removing from queue
+                        Binding b = new Binding(destinationDB, DestinationType.QUEUE, exchangeDB, mongoDbBindings, null);
+                        amqpAdmin = new RabbitAdmin(connectionFactory());
+                        amqpAdmin.removeBinding(b);
+                        // Removing binding document from mongoDB
+                        mongoDBHandler.dropDocument(bindingKeysDataBaseName, bindingKeyCollectionName, condition);
+                    }
+                    else{
+                        // storing the existing key into an array.
+                        existingBindingsData.add(mongoDbBindings);
+                    }
+            }
+        }
+        // comparing with the stored key and adding the new binding key into the mongoDB.
+        for(final Binding bindingKey:bindingList){
```

Comment:
This part of the code that store newBindingKeys into MongoDb, could be put to an own private function where the function name says what it do:

private void storeNewBindingKeys(ArrayList<> AMQPBindingObjectList)

or something.